### ISSUE TITLE
HDFStore fails to read non-ascii characters

### ISSUE THREAD

**Issue by FilipDusek** [pydata/pandas#11234]
When I try to save some non-ascii character like é and then load it again, I end up with UnicodeDecodeError. If you add some more data to the string (like 'aée'), the data gets stored and retrieved without error, but the result is missing the last character.

```python
import pandas as pd

df = pd.DataFrame(columns=["A"])
toAppend = {"A": "é"}
df = df.append(toAppend, ignore_index = True)

store = pd.HDFStore(r'thiswillcrash.h5')
store.put('df', df, format='table', encoding="utf-8")
d = store["df"]
print(d)

store.close()
```
Versions
```
INSTALLED VERSIONS
------------------
commit: None
python: 3.4.3.final.0
python-bits: 64
OS: Windows
OS-release: 8
machine: AMD64
processor: Intel64 Family 6 Model 69 Stepping 1, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None

pandas: 0.16.2
nose: 1.3.4
Cython: 0.22
numpy: 1.9.3
scipy: 0.15.1
statsmodels: 0.6.1
IPython: 3.0.0
sphinx: 1.2.3
patsy: 0.3.0
dateutil: 2.4.2
pytz: 2015.6
bottleneck: None
tables: 3.2.2
numexpr: 2.4.4
matplotlib: 1.4.3
openpyxl: 1.8.5
xlrd: 0.9.3
xlwt: None
xlsxwriter: 0.6.7
lxml: 3.4.4
bs4: 4.3.2
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: 0.9.9
pymysql: None
psycopg2: None
```

**Comment by jreback** [comment]
should be fixed by : https://github.com/pydata/pandas/pull/10889

give a try with ``v0.17.0rc2``

``conda install pandas -c pandas``

**Comment by FilipDusek** [comment]
No, unfortunately I still get the error

```python
INSTALLED VERSIONS
------------------
commit: None
python: 3.4.3.final.0
python-bits: 64
OS: Windows
OS-release: 8
machine: AMD64
processor: Intel64 Family 6 Model 69 Stepping 1, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None

pandas: 0.17.0rc2
nose: 1.3.7
pip: 7.1.2
setuptools: 18.3.2
Cython: 0.22.1
numpy: 1.9.3
scipy: 0.15.1
statsmodels: 0.6.1
IPython: 3.2.0
sphinx: 1.3.1
patsy: 0.3.0
dateutil: 2.4.2
pytz: 2015.6
blosc: None
bottleneck: 1.0.0
tables: 3.2.0
numexpr: 2.4.3
matplotlib: 1.4.3
openpyxl: 1.8.5
xlrd: 0.9.3
xlwt: 1.0.0
xlsxwriter: 0.7.3
lxml: 3.4.4
bs4: 4.3.2
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: 1.0.5
pymysql: None
psycopg2: None
```

**Comment by TomAugspurger** [comment]
@jreback looks like we truncate the column to be length 1 since `len(df.iloc[0, 0])` is 1.

This works though

```python
In [19]: df = pd.DataFrame({'A': ['é']})

In [20]: store = pd.HDFStore(r'thiswillcrash.h5')

In [21]: store.put('df', df, format='table', min_itemsize={'A': 30})

In [22]: store.get('df')
Out[22]:
   A
0  é
```

Do you have a good idea where a fix would go?

**Comment by jreback** [comment]
https://github.com/pydata/pandas/blob/master/pandas/lib.pyx#L972

is where the width of the strings are determined 
but it should work for unicode

**Comment by TomAugspurger** [comment]
Is it because the encoded length is different than the number of characters?

```python
In [10]: x
Out[10]: 'é'

In [11]: len(x)
Out[11]: 1

In [12]: len(x.encode('utf-8'))
Out[12]: 2
```

**Comment by jreback** [comment]
yep should encode before we check and set the length

**Comment by jreback** [comment]
closed  by #11240
